On big-endian machines, a test case looking for the methods _L1D and _E1D
in the same scope would fail (see tests/misc/badcode.asl:184).  The names
to be compared were being treated as 32-bit ints, and not strings.  Hence,
the characters were re-ordered incorrectly, mismatching the assumptions
made in the remainder of the function.

Index: acpica-unix-20140724/source/compiler/aslanalyze.c
===================================================================
--- acpica-unix-20140724.orig/source/compiler/aslanalyze.c
+++ acpica-unix-20140724/source/compiler/aslanalyze.c
@@ -446,7 +446,7 @@ ApCheckForGpeNameConflict (
 
     /* Need a null-terminated string version of NameSeg */
 
-    ACPI_MOVE_32_TO_32 (Name, &Op->Asl.NameSeg);
+    ACPI_MOVE_NAME (Name, &Op->Asl.NameSeg);
     Name[ACPI_NAME_SIZE] = 0;
 
     /*
@@ -473,7 +473,7 @@ ApCheckForGpeNameConflict (
      * We are now sure we have an _Lxx or _Exx.
      * Create the target name that would cause collision (Flip E/L)
      */
-    ACPI_MOVE_32_TO_32 (Target, Name);
+    ACPI_MOVE_NAME (Target, Name);
 
     /* Inject opposite letter ("L" versus "E") */
 
